src/city/finance.c

```diff
@@ -346,6 +348,15 @@ static void pay_monthly_building_levies(void)
         }
     }
 
+    int grid_offset = map_data.start_offset;
+    for (int y = 0; y < map_data.height; y++, grid_offset += map_data.border_size) {
+        for (int x = 0; x < map_data.width; x++, grid_offset++) {
+            if (map_terrain_is(grid_offset, TERRAIN_HIGHWAY_TOP_LEFT)) {
```

Comment:
Are you positively sure that 1/4 of the tiles are of the top left variant?

As an alternative, you can count all the highway tiles and divide the result by 4.

Reply:
I've gone to great lengths to ensure that highways can only be added and removed in 2x2 groups (mostly to ensure players can't do anything sneaky to get a 1 tile wide highway). If they overlap, the terrains are ORed together.